src/thor/triplegbuilder.cc

```diff
@@ -1138,8 +1138,15 @@ void TripLegBuilder::Build(
   // Get the first nodes graph id by using the end node of the first edge to get the tile with the
   // opposing edge then use the opposing index to get the opposing edge, and its end node is the
   // begin node of the original edge
-  auto* first_edge = graphreader.GetGraphTile(path_begin->edgeid)->directededge(path_begin->edgeid);
+  auto begin_tile = graphreader.GetGraphTile(path_begin->edgeid);
+  if (begin_tile == nullptr) {
+    return; // or throw?
```

Comment:
`throw` everything in triplegbuilder has to be a throw. at this point in the computation you have the route. if you cant get the tiles along it then you cant proceed to serialize a route.